Taweret/mix/linear.py

```diff
@@ -157,12 +158,21 @@ def plot_weights(self, mixture_params : np.ndarray, x : np.ndarray) -> np.ndarra
             input parameter values
         """
 
-        fig, ax = plt.subplots()
-        ax.plot(x, mixture_function(self.method, x, mixture_params), label=self.method)
-        ax.set_ylabel('Weights')
-        ax.set_xlabel('Input Parameter')
+        if self.method != 'dirchlet' and self.method != 'beta':
+            fig, ax = plt.subplots()
+            weights = mixture_function(self.method, x, mixture_params)
+            ax.plot(x, weights[0], label=self.method)
+            ax.legend()
+            ax.set_ylabel('Weights')
+            ax.set_xlabel('Input Parameter')
+        else:
```

Comment:
Why do we need separate plotting for 'dirchlet' and ' beta' methods?

Reply:
Note that what I plot for `beta` and `dirichlet` are scatter plots, `weights` vs `model number`.
For these cases, the `mixture_function` returns random variables that are independent of the model parameter `x`.
Therefore, it does not make sense to plot it as a function of `x`.

Reply:
So this mixing is a global mixing of models? I think the goal is to do a local mixing. Vojtech had some slides about this in his presentation.

Reply:
I see, yes and we spoke about this with Uli on Monday.
He thinks that using a GP for the prior is reasonable.
I somehow do not have access to the slides on this computer right now, but I will look at it again when I am at school.

Reply:
ahh I see. I think then we might have to be more careful. We are trying to work with local and global mixing methods in the same mixing method (linear.py). Now I think the modification you made for log_likelihood might have an error. I will open an issue there.